spec/controllers/admin/users_controller_spec.rb

```diff
@@ -23,7 +25,27 @@
       end
       its(:status){ should == 200 }
     end
+
+    context "when I have 50 users" do
+      before do
+        create_list(:user, 50 - User.count)
```

Comment:
Num sei se ta ligado, mas teoricamente a cada teste (`it`), seu banco é zerado.
Você já testou só com o 50, e realmente da problema no seu teste?
Se sim, o que ta fudendo mesmo é o `before` que me mostrou. Para outra pessoa afetar seu teste ela teria que criar um novo usuário dentro do mesmo `before` do current_user la.
Mas enfim se ta dando problema mesmo, deixa assim.